### ISSUE TITLE
[form-builder] -- Setting a message on field over writes all error messages

### ISSUE THREAD

**Issue by edster** [pyrocms/pyrocms#4742]
Given the below example,  `'messages'     => ['required' => 'You must accept the Terms & Conditions.',],` will over write ALL error messages. So if you have anouther required field, say phone, if you submit the form, it will ALWAYS only show `'You must accept the Terms & Conditions.',`. Even if the `accept` field is checked off.

```
$fields['accept'] = [
            'required'     => TRUE,
            'label'        => FALSE,
            'save'         => FALSE,
            'type'         => 'anomaly.field_type.boolean',
            'messages'     => [
                'required' => 'You must accept the Terms & Conditions.',
            ],
            'config'       => [
                'mode'  => 'checkbox',
                'label' => 'I Accept the Terms & Conditions',
            ]
```

**Comment by edster** [comment]
So trying to work through this, I found that `FormMessages::class` is iterating through all fields and looking for custom messages, if it finds them, is trying to set them. However it is just setting the `$rule -> message`

This ends up just returning the last custom message combination.

I would think you would want to have an array of fields->[rule->message]
` $messages[$field->field] = [$rule => $message];`

Which would end up liek
```
array:3 [▼
  "name" => array:1 [▼
    "required" => "You must include a your name"
  ]
  "phone" => array:1 [▼
    "required" => "You must include a phone number"
  ]
  "accept" => array:1 [▼
    "required" => "You must accept the TOS"
  ]
]
```
vs
```
array:1 [▼
    "required" => "You must accept the TOS"
]
```

**Comment by edster** [comment]
Fixed!

https://github.com/anomalylabs/streams-platform/pull/539

**Comment by edster** [comment]
Merged into 1.3
